### PR TITLE
New WANDPowderReduction Algorithm

### PR DESCRIPTION
**Pull request by rosswhitfield**
This algorithm will do powder diffraction reduction for WAND².

**To test:**
Try the usage examples.

Release Notes added.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                              MaskAngle, ExtractMask, Minus,
+                              RemoveMaskedSpectra, mtd)
+from mantid.kernel import StringListValidator, Direction, Property
+
+
+class WANDPowderReduction(DataProcessorAlgorithm):
+    temp_workspace_list = ['__data_tmp', '__cal_tmp', '__mask_tmp', '__bkg_tmp']
+
+    def category(self):
+        return "Diffraction\\Reduction"
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py]: [Python raw string](https://docs.python.org/2.0/ref/strings.html) would work as well. Doesn't need to be changed unless you really want to.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                bkg_scale = bkg.run().getProtonCharge()
+            elif normaliseBy == "Time":
+                bkg_scale = bkg.run().getLogData('duration').value
+            Scale(InputWorkspace='__bkg_tmp', OutputWorkspace='__bkg_tmp', Factor=cal_scale/bkg_scale, EnableLogging=False)
+            Minus(LHSWorkspace=outWS, RHSWorkspace='__bkg_tmp', OutputWorkspace=outWS, EnableLogging=False)
+
+        self.setProperty("OutputWorkspace", outWS)
+
+        # remove temp workspaces
+        [DeleteWorkspace(ws, EnableLogging=False) for ws in self.temp_workspace_list if mtd.doesExist(ws)]
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py]: Nice! You might want to leave this as a good example for other people to follow in slack.

**Review by peterfpeterson — APPROVED** [review]
Looks good. The only suggestion (for a later ticket) is to add a workflow diagram ([example](https://github.com/mantidproject/mantid/blob/master/docs/source/diagrams/AlignAndFocusPowder-v1_wkflw.dot)) which may help users' understanding, but the source is pretty clear on its own.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py (hunk 1 above)
Comment (rosswhitfield) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/WANDPowderReduction.py]: This is consistent with every other algorithm.

**Comment by rosswhitfield** [comment]
I will work on the workflow diagram and will include it in the next PR adding more options to this algorithm.

**Pull request merged**